### PR TITLE
Support for large unsigned integer constants up to 256 bits

### PR DESCRIPTION
**Pull request by jeanmon**
Currently a literal unsigned integer larger than 32 bits is not supported.

We suffix unsigned integer literals below 64 bits with the "UL" suffix.

Integers with more than 64 bits and less than 256 is supported through a uint256_t constructor call.

### CODE REVIEW

File: bberg/src/relation_builder.rs [hunk 1]
```diff
@@ -328,7 +329,27 @@ fn craft_expression<T: FieldElement>(
     collected_public_identities: &mut HashSet<String>,
 ) -> BBIdentity {
     match expr {
-        Expression::Number(n) => (1, format!("FF({})", n.to_arbitrary_integer())),
+        Expression::Number(n) => {
```
Comment (Maddiaa0) [inline bberg/src/relation_builder.rs:332]: Could all of these be covered by the FF("hex string") constructor ?

File: bberg/src/relation_builder.rs [hunk 2]
```diff
@@ -328,7 +329,27 @@ fn craft_expression<T: FieldElement>(
...
+            let number: BigUint = n.to_arbitrary_integer();
+            if number.bits() < 32 {
+                return (1, format!("FF({})", number));
+            }
+            if number.bits() < 64 {
+                return (1, format!("FF({}UL)", number));
+            }
+            if number.bits() < 256 {
+                let mut chunks: Vec<u64> = number.iter_u64_digits().collect::<Vec<u64>>();
+                chunks.resize(4, 0);
```
Comment (Maddiaa0) [inline bberg/src/relation_builder.rs:342]: Is the endianness of this concrete ?

**Review by Maddiaa0 — APPROVED** [review]
Nice

File: bberg/src/relation_builder.rs (hunk 2 above)
Comment (jeanmon) [reply bberg/src/relation_builder.rs:342]: You meant "correct"? Yes I checked this.

File: bberg/src/relation_builder.rs (hunk 1 above)
Comment (jeanmon) [reply bberg/src/relation_builder.rs:332]: Good idea. According to the current cpp source code, this constructor takes only zero-padded/fixed size hex representation. This means that the cpp generated code would have very long string even for small constants which is not so nice I would say.

See relevant cpp source code:
https://github.com/AztecProtocol/aztec-packages/blob/f1eb6d5b6e3221e5a5b8f55632dba066d4cfbc24/barretenberg/cpp/src/barretenberg/numeric/uint256/uint256.hpp#L61

**Pull request merged**